tests/unit_tests/test_tree.py

```diff
@@ -39,5 +43,105 @@ async def test_parsed_tree_empty_json():
         future.set_result(mocked_response)
         mocked_client.get.return_value = future
 
-        data = await parsed_tree(mocked_client, root_node_id)
+        data = await tree_from_vocabs(mocked_client, root_node_id)
         assert len(data) == 0
+
+
+def test_parse_tree():
+    node_id_biology = uuid.UUID("15fce411-54d9-467f-8f35-61ea374a298d")
+
+    query = tree_query(node_id_biology)
+    expected_query = {
+        "_source": [
+            "nodeRef.id",
+            "properties.cm:title",
+            "collections.path",
+            "parentRef.id",
+        ],
+        "from": 0,
+        "query": {
+            "bool": {
+                "filter": [{"term": {"path": node_id_biology}}],
+                "must": [
+                    {"match": {"permissions.Read": "GROUP_EVERYONE"}},
+                    {"match": {"properties.cm:edu_metadataset": "mds_oeh"}},
+                    {"match": {"nodeRef.storeRef.protocol": "workspace"}},
+                ],
+            }
+        },
+        "size": 500000,
+        "sort": ["fullpath"],
+    }
+
+    assert query.to_dict() == expected_query
+
+    if os.getenv("CI", False):
+        directory = "tests/unit_tests/resources"
+    else:
+        directory = "unit_tests/resources"
+
+    with open(f"{directory}/test_tree.json") as file:
+        expected_tree = json.loads("".join(file.readlines()))
+    with open(f"{directory}/test_tree_response.json") as file:
+        response = json.loads("".join(file.readlines()))
+        response = [hit["_source"] for hit in response]
+
+    with mock.patch("app.api.collections.tree.Search.execute") as mocked_execute:
```

Comment:
from here to the end, i have absolutely no clue what the test does and is supposed to check :-/

Can't we 
 - use a (significantly reduced) reponse loaded from json. E.g. simple tree depth 3 and non trivial structure.
 - Then hardcode the expected internal representation (I assume that would be the nested structure of `CollectionNode` instances)
 - Then compare the parsed and constructed nested structure with the hardcoded expectation. It should be possible to just compare equality when we used pydantic BaseModel.

Reply:
I reduced the response and the expected tree to few nodes only.
Regarding the equality, currently I do not compare for the specific nodes, but their structure relative to each other, i.e., how many nodes are there and how many of them are parent nodes. I could change it to compare node by node - as you prefer.
I reduced the amount of asserts meaningfully and reduced intention hiding.